{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib\n",
    "matplotlib.style.use(\"ggplot\")\n",
    "import pandas as pd\n",
    "import random as rand\n",
    "import string\n",
    "import math\n",
    "chars = string.ascii_uppercase + string.digits\n",
    "%matplotlib inline\n",
    "plt.rcParams[\"figure.figsize\"] = (16,8)\n",
    "import seaborn as sns\n",
    "import sklearn as skl\n",
    "import skimage as ski\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "from scipy import spatial\n",
    "from sklearn.utils import shuffle\n",
    "from os import system\n",
    "import skimage as ski\n",
    "from skimage.color import rgb2gray\n",
    "from scipy import misc\n",
    "from skimage.exposure import adjust_gamma\n",
    "from sklearn import preprocessing\n",
    "from scipy import stats\n",
    "from glob import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_list = pd.read_csv('allDiffexpGenes.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(356, 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>qval</th>\n",
       "      <th>up_or_down</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FBgn0003071</td>\n",
       "      <td>0.000089</td>\n",
       "      <td>up</td>\n",
       "      <td>Pfk</td>\n",
       "      <td>Catalyzes the phosphorylation of D-fructose 6-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FBgn0028426</td>\n",
       "      <td>0.000458</td>\n",
       "      <td>down</td>\n",
       "      <td>JhI-1</td>\n",
       "      <td>Juvenile hormone-inducible protein 1 is a high...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FBgn0013679</td>\n",
       "      <td>0.001081</td>\n",
       "      <td>up</td>\n",
       "      <td>mt:ND1</td>\n",
       "      <td>Core subunit of the mitochondrial membrane res...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FBgn0013680</td>\n",
       "      <td>0.001081</td>\n",
       "      <td>up</td>\n",
       "      <td>mt:ND2</td>\n",
       "      <td>The NADH:ubiquinone oxidoreductase complex (Mi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FBgn0013681</td>\n",
       "      <td>0.001081</td>\n",
       "      <td>up</td>\n",
       "      <td>mt:ND3</td>\n",
       "      <td>oxidation-reduction process.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene     qval  up_or_down gene_name  \\\n",
       "0  FBgn0003071  0.000089         up       Pfk   \n",
       "1  FBgn0028426  0.000458       down     JhI-1   \n",
       "2  FBgn0013679  0.001081         up    mt:ND1   \n",
       "3  FBgn0013680  0.001081         up    mt:ND2   \n",
       "4  FBgn0013681  0.001081         up    mt:ND3   \n",
       "\n",
       "                                         description  \n",
       "0  Catalyzes the phosphorylation of D-fructose 6-...  \n",
       "1  Juvenile hormone-inducible protein 1 is a high...  \n",
       "2  Core subunit of the mitochondrial membrane res...  \n",
       "3  The NADH:ubiquinone oxidoreductase complex (Mi...  \n",
       "4                       oxidation-reduction process.  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(gene_list.shape)\n",
    "gene_list.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_list_upregulated = gene_list[gene_list[\"up_or_down\"] == 'up']\n",
    "gene_list_downregulated = gene_list[gene_list[\"up_or_down\"] == 'down']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(192, 5)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_list_downregulated.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(164, 5)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_list_upregulated.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_names_upregulated = gene_list_upregulated.gene_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_names_upregulated.to_csv('Upregulated genes names.csv', index = False)\n",
    "gene_list_upregulated.gene.to_csv('Upregulated genes code.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_names_downregulated = gene_list_downregulated.gene_name\n",
    "gene_names_downregulated.to_csv('Downregulated genes names.csv', index = False)\n",
    "gene_list_downregulated.gene.to_csv('Downregulated genes code.csv', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# used table browser from UCSC to obtain the bed files for upregulated and downregulated genes \n",
    "\n",
    "https://ampliseq.com/help/bedFilesUcsc.action\n",
    "\n",
    "it dropped a few genes for some reason\n",
    "\n",
    "then made a image of the chromossomes positions using https://www.ncbi.nlm.nih.gov/genome/tools/gdp\n",
    "\n",
    "for some weird reason the number of genes jumped from ~180 to 381 in both cases. Need to understand why that happened\n",
    "\n",
    "Changed the header of the bed file cause it's a mess and lacked the column names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(381, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>gene</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>thickStart</th>\n",
       "      <th>thickEnd</th>\n",
       "      <th>itemRgb</th>\n",
       "      <th>blockCount</th>\n",
       "      <th>blockSizes</th>\n",
       "      <th>blockStarts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>chr2L</td>\n",
       "      <td>7018086</td>\n",
       "      <td>7023898</td>\n",
       "      <td>NM_001014477</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>7018144</td>\n",
       "      <td>7023705</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>288,650,248,370,</td>\n",
       "      <td>0,344,1048,5442,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>chr2R</td>\n",
       "      <td>24990947</td>\n",
       "      <td>25011463</td>\n",
       "      <td>NM_001014552</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>24991136</td>\n",
       "      <td>25011341</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>325,164,1296,531,305,2740,93,24,122,157,216,23...</td>\n",
       "      <td>0,862,1085,2443,3064,3459,6261,7330,9091,11340...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>chr2R</td>\n",
       "      <td>24990947</td>\n",
       "      <td>25011463</td>\n",
       "      <td>NM_001014553</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>24991136</td>\n",
       "      <td>25010511</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>325,164,1296,531,305,2740,93,24,122,157,216,23...</td>\n",
       "      <td>0,862,1085,2443,3064,3459,6261,7330,9091,11340...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>chr3R</td>\n",
       "      <td>25983012</td>\n",
       "      <td>25995737</td>\n",
       "      <td>NM_001014668</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>25983291</td>\n",
       "      <td>25994574</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>336,144,385,142,594,322,553,538,1592,607,2356,...</td>\n",
       "      <td>0,579,822,1262,1460,2115,2499,3106,3699,5347,6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chrX</td>\n",
       "      <td>16361568</td>\n",
       "      <td>16368526</td>\n",
       "      <td>NM_001014747</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>16361697</td>\n",
       "      <td>16365744</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>486,1273,309,2940,</td>\n",
       "      <td>0,1789,3636,4018,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chromosome     start      stop          gene  score strand  thickStart  \\\n",
       "74       chr2L   7018086   7023898  NM_001014477      0      -     7018144   \n",
       "122      chr2R  24990947  25011463  NM_001014552      0      -    24991136   \n",
       "123      chr2R  24990947  25011463  NM_001014553      0      -    24991136   \n",
       "293      chr3R  25983012  25995737  NM_001014668      0      +    25983291   \n",
       "18        chrX  16361568  16368526  NM_001014747      0      +    16361697   \n",
       "\n",
       "     thickEnd  itemRgb  blockCount  \\\n",
       "74    7023705        0           4   \n",
       "122  25011341        0          13   \n",
       "123  25010511        0          13   \n",
       "293  25994574        0          18   \n",
       "18   16365744        0           4   \n",
       "\n",
       "                                            blockSizes  \\\n",
       "74                                    288,650,248,370,   \n",
       "122  325,164,1296,531,305,2740,93,24,122,157,216,23...   \n",
       "123  325,164,1296,531,305,2740,93,24,122,157,216,23...   \n",
       "293  336,144,385,142,594,322,553,538,1592,607,2356,...   \n",
       "18                                  486,1273,309,2940,   \n",
       "\n",
       "                                          blockStarts   \n",
       "74                                    0,344,1048,5442,  \n",
       "122  0,862,1085,2443,3064,3459,6261,7330,9091,11340...  \n",
       "123  0,862,1085,2443,3064,3459,6261,7330,9091,11340...  \n",
       "293  0,579,822,1262,1460,2115,2499,3106,3699,5347,6...  \n",
       "18                                   0,1789,3636,4018,  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed_upregal = pd.read_csv('upregul_bed.csv', sep= \"\t\")\n",
    "print(bed_upregal.shape)\n",
    "bed_upregal.sort_values('gene').head()\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(381, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>gene</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>thickStart</th>\n",
       "      <th>thickEnd</th>\n",
       "      <th>itemRgb</th>\n",
       "      <th>blockCount</th>\n",
       "      <th>blockSizes</th>\n",
       "      <th>blockStarts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>529</th>\n",
       "      <td>chr2R</td>\n",
       "      <td>24903415</td>\n",
       "      <td>24903935</td>\n",
       "      <td>NM_001014551</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>24903722</td>\n",
       "      <td>24903892</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>174,91,55,</td>\n",
       "      <td>0,282,465,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>chr2L</td>\n",
       "      <td>21419530</td>\n",
       "      <td>21419943</td>\n",
       "      <td>NM_001032113</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>21419575</td>\n",
       "      <td>21419887</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>413,</td>\n",
       "      <td>0,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>chr2L</td>\n",
       "      <td>21439588</td>\n",
       "      <td>21440001</td>\n",
       "      <td>NM_001032113</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>21439633</td>\n",
       "      <td>21439945</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>413,</td>\n",
       "      <td>0,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>chr2L</td>\n",
       "      <td>21453789</td>\n",
       "      <td>21454202</td>\n",
       "      <td>NM_001032113</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>21453834</td>\n",
       "      <td>21454146</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>413,</td>\n",
       "      <td>0,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>chr2L</td>\n",
       "      <td>21484084</td>\n",
       "      <td>21484497</td>\n",
       "      <td>NM_001032113</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>21484129</td>\n",
       "      <td>21484441</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>413,</td>\n",
       "      <td>0,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chromosome     start      stop          gene  score strand  thickStart  \\\n",
       "529      chr2R  24903415  24903935  NM_001014551      0      -    24903722   \n",
       "50       chr2L  21419530  21419943  NM_001032113      0      -    21419575   \n",
       "53       chr2L  21439588  21440001  NM_001032113      0      -    21439633   \n",
       "54       chr2L  21453789  21454202  NM_001032113      0      -    21453834   \n",
       "55       chr2L  21484084  21484497  NM_001032113      0      -    21484129   \n",
       "\n",
       "     thickEnd  itemRgb  blockCount  blockSizes blockStarts   \n",
       "529  24903892        0           3  174,91,55,   0,282,465,  \n",
       "50   21419887        0           1        413,           0,  \n",
       "53   21439945        0           1        413,           0,  \n",
       "54   21454146        0           1        413,           0,  \n",
       "55   21484441        0           1        413,           0,  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed_downregal = pd.read_csv('downregul_bed.csv', sep= \"\t\")\n",
    "print(bed_upregal.shape)\n",
    "bed_downregal.sort_values('gene').head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "bed_upregal.iloc[:, :3].to_csv('upregulated_Chr_start_Stop.csv', index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "bed_downregal.iloc[:, :3].to_csv('downregulated_Chr_start_Stop.csv', index= False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# opening file from rVista \n",
    "\n",
    "http://genome.lbl.gov/vista/downloads.shtml\n",
    "\n",
    "dmel_dvir.gz\t- transcription factor binding sites in promoters (5K upstream)\n",
    "                      of D.melanogaster Release 5 FlyBase genes conserved\n",
    "                      in D.melanogaster-D.virilis alignments\n",
    "                      \n",
    "                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "tf_binding = pd.read_csv('bed files/dmel_dvir',header = None, sep= \"\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2018959, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FBgn0000053</td>\n",
       "      <td>Abd-B</td>\n",
       "      <td>chr2L</td>\n",
       "      <td>7024253</td>\n",
       "      <td>7024259</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FBgn0000053</td>\n",
       "      <td>Abd-B</td>\n",
       "      <td>chr2L</td>\n",
       "      <td>7024257</td>\n",
       "      <td>7024263</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FBgn0000053</td>\n",
       "      <td>Abd-B</td>\n",
       "      <td>chr2L</td>\n",
       "      <td>7024916</td>\n",
       "      <td>7024922</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FBgn0000053</td>\n",
       "      <td>Abd-B</td>\n",
       "      <td>chr2L</td>\n",
       "      <td>7025928</td>\n",
       "      <td>7025934</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FBgn0000053</td>\n",
       "      <td>Abd-B</td>\n",
       "      <td>chr2L</td>\n",
       "      <td>7025931</td>\n",
       "      <td>7025937</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             0      1      2        3        4  5\n",
       "0  FBgn0000053  Abd-B  chr2L  7024253  7024259  +\n",
       "1  FBgn0000053  Abd-B  chr2L  7024257  7024263  +\n",
       "2  FBgn0000053  Abd-B  chr2L  7024916  7024922  +\n",
       "3  FBgn0000053  Abd-B  chr2L  7025928  7025934  -\n",
       "4  FBgn0000053  Abd-B  chr2L  7025931  7025937  +"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(tf_binding.shape)\n",
    "tf_binding.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# find bindings for upregulated and downregulated genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(20476, 6)"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tf_down = tf_binding[tf_binding[0].isin(gene_list_downregulated.gene.values)]\n",
    "tf_down.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(19743, 6)"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tf_up = tf_binding[tf_binding[0].isin(gene_list_upregulated.gene.values)]\n",
    "tf_up.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Find unique TFs and count their frequencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CF2-II</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BR-C</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BYN</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CF1A</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Antp</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>TWI</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Sn</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Hairy</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Hb</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Abd-B</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SGF-3</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Elf-1</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>DEAF1</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>dl</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>En</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Ovo</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Adf-1</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Zeste</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Kr</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCM</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>E74A</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>mtTFA</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>DREF</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HSF</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Croc</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>OVO</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Grainyhead/Elf-1/NTF</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CF1</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Bcd</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CI</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Ttk</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Ftz</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>STAT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Sry-beta</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       0   1\n",
       "8                 CF2-II  79\n",
       "3                   BR-C  79\n",
       "4                    BYN  79\n",
       "7                   CF1A  79\n",
       "2                   Antp  78\n",
       "29                   TWI  77\n",
       "27                    Sn  77\n",
       "20                 Hairy  77\n",
       "21                    Hb  76\n",
       "0                  Abd-B  75\n",
       "25                 SGF-3  72\n",
       "14                 Elf-1  69\n",
       "11                 DEAF1  68\n",
       "32                    dl  60\n",
       "15                    En  60\n",
       "24                   Ovo  55\n",
       "1                  Adf-1  51\n",
       "31                 Zeste  48\n",
       "22                    Kr  45\n",
       "17                   GCM  37\n",
       "13                  E74A  34\n",
       "33                 mtTFA  34\n",
       "12                  DREF  32\n",
       "19                   HSF  31\n",
       "10                  Croc  24\n",
       "23                   OVO  21\n",
       "18  Grainyhead/Elf-1/NTF  20\n",
       "6                    CF1  19\n",
       "5                    Bcd   8\n",
       "9                     CI   7\n",
       "30                   Ttk   5\n",
       "16                   Ftz   4\n",
       "26                  STAT   1\n",
       "28              Sry-beta   1"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tf_down_names = tf_down.groupby(1)[0].nunique()\n",
    "tf_down_names.to_csv('tf genes binding to downregulated genes')\n",
    "tf_down_namesT = pd.read_csv('tf genes binding to downregulated genes', header = None)\n",
    "tf_down_namesT.sort_values(1, ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CF2-II</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BR-C</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BYN</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CF1A</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Abd-B</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>TWI</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Antp</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Sn</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Hb</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Hairy</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SGF-3</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>En</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>DEAF1</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Elf-1</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>dl</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Adf-1</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Ovo</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Kr</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Zeste</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCM</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>E74A</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>DREF</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HSF</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>mtTFA</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>OVO</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Croc</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Grainyhead/Elf-1/NTF</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CF1</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Bcd</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Ftz</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CI</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>STAT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Sry-beta</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Ttk</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       0   1\n",
       "8                 CF2-II  71\n",
       "3                   BR-C  71\n",
       "4                    BYN  71\n",
       "7                   CF1A  70\n",
       "0                  Abd-B  69\n",
       "29                   TWI  69\n",
       "2                   Antp  68\n",
       "27                    Sn  68\n",
       "21                    Hb  67\n",
       "20                 Hairy  63\n",
       "25                 SGF-3  62\n",
       "15                    En  59\n",
       "11                 DEAF1  55\n",
       "14                 Elf-1  54\n",
       "32                    dl  53\n",
       "1                  Adf-1  46\n",
       "24                   Ovo  45\n",
       "22                    Kr  42\n",
       "31                 Zeste  38\n",
       "17                   GCM  33\n",
       "13                  E74A  30\n",
       "12                  DREF  29\n",
       "19                   HSF  24\n",
       "33                 mtTFA  24\n",
       "23                   OVO  23\n",
       "10                  Croc  20\n",
       "18  Grainyhead/Elf-1/NTF  19\n",
       "6                    CF1  14\n",
       "5                    Bcd  12\n",
       "16                   Ftz   6\n",
       "9                     CI   5\n",
       "26                  STAT   1\n",
       "28              Sry-beta   1\n",
       "30                   Ttk   1"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tf_up_names = tf_up.groupby(1)[0].nunique()\n",
    "tf_up_names.to_csv('tf genes binding to upregulated genes')\n",
    "tf_up_namesT = pd.read_csv('tf genes binding to upregulated genes', header = None)\n",
    "tf_up_namesT.sort_values(1, ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(34, 2)"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
